Newsletters: add 142 (2021-03-27)

harding opened the pull request:
- [x] Update releases/RCs @harding 
- [x] Add topic links @harding 
- [x] StackExchange section @bitschmidty 
- [x] `Bitcoin Core #17227` Qt: Add Android packaging support @jnewbery
- [x] `Eclair #1738`` Handle aggregated anchor outputs htlc txs @dongcarl

Added by John:

- [x] Include link to https://bitcoinops.org/en/payment-batching/. I think it's a great resource, so we should let people know about it!
- [x] Review and merge #535. From Harding in that PR: "I'd like to get this merged by Tuesday afternoon so that we can link to it in next Wednesday's newsletter, which mentions dual funding receiving a C-Lightning run-time configuration option."

jnewbery commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,134 @@
+   existing algorithms use a heuristic approach.  Combined with JIT
+   rebalancing, they estimate a 48% improvement.  Given that each retry
+   usually requires several seconds, and could take much longer in some
+   cases, this could provide a significant benefit to the network.
```
I'm not sure what "benefit to the network" means here. Would "improved user experience" be better or is it broader than that?

jnewbery commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,134 @@
+
+- [Rust-Lightning #849][] makes a channel's `cltv_expiry_delta`
+  configurable and reduces the default value from 72 blocks to 36
+  blocks.  This parameter sets the minimum number of blocks a node will
+  have to settle a payment attempt with its upstream peer after learning
+  from its downstream peer whether or not that payment succeeded; it must
```
"This parameter [...] payment succeeded;" is a bit long and difficult to parse. "the minimum number of blocks a node will have to settle" in particular reads like the node has to settle blocks, whereas it really means "the number of blocks within which a node must settle".

jnewbery commented:
Added #17227.

jonatack commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,142 @@
+   one drawn from a snapshot of almost 1,000 live channels.
+
+   The paper deliberately does not take routing fees into consideration,
+   and most mailing list discussion about the topic focused on how to
```
perhaps: "most of the ensuing mailing list discussion focused on..."

jonatack commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,142 @@
+
+- [C-Lightning 0.10.0-rc1][C-Lightning 0.10.0] is a release candidate
+  for the next major version of this LN node software.
+
```
Just released, if interesting:
- Specter desktop 1.3.0 https://github.com/cryptoadvance/specter-desktop/releases/tag/v1.3.0

jonatack replied:
this one might be important 

https://github.com/btcpayserver/btcpayserver/releases/tag/v1.0.7.1

"This is a security release that patches one critical and several low-impact vulnerabilities that affected BTCPay Server versions 1.0.7.0 and older."

harding replied:
@jonatack indeed!  Thanks!

jonatack reviewed (commented):
First quick look at the probabilistic path selection paper section only -- looks good.

jonatack commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,177 @@
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [How hard is it for an exchange to adopt native segwit?]({{bse}}103674)
+  instagibbs lists a few considerations for exchanges implementing native segwit
```
```suggestion
  instagibbs lists a few considerations for exchanges implementing native segwit,
```

jonatack commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,177 @@
+- [How hard is it for an exchange to adopt native segwit?]({{bse}}103674)
+  instagibbs lists a few considerations for exchanges implementing native segwit
+  including address generation, compatibility of signing infrastructure like HSMs,
+  ensuring spendability, and support and business considerations.
```
it's unclear (to me) if the "ensuring..." part is in the same list or a subordinate clause to "like HSMs"

"HSMs" doesn't seem to be an unintroduced abbreviation

harding replied:
@bitschmidty regarding the "ensuring..." feedback, one of the writer tricks that mostly avoids this problem is to always put list items in order of least to most number of syllables when no better sort order suggests itself.  E.g.: "address generation, ensuring spendability, support and business considerations, and compatibility of signing infrastructure like Hardware Security Modules (HSMs)."

(One exception to this rule is to sort most to least number of syllables when you want your list to end memorably, suddenly, and sharply.)

bitschmidty replied:
@jonatack thanks for pointing this out and thanks @harding for getting me out of my bind! Took the suggestion

jonatack commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,177 @@
+  ensuring spendability, and support and business considerations.
+
+- [How do you calculate when 98% of Bitcoin will be mined?]({{bse}}103159)
+  While Murch provides an approximation of either 2030 or 2031 for when 98% of
```
maybe
```suggestion
  While Murch provides an estimate of 2030-2031 for when 98% of
```

jonatack commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,177 @@
+  Time Past (MTP) and block heights as timing mechanisms within Bitcoin. MTP
+  roughly corresponds to clock time but can manipulated by miners to skip a
+  signaling period. Block heights are not consistent but are also not
+  miner-gamable in the same way as MTP.
```
```suggestion
  miner-gameable in the same way as MTP.
```

dongcarl commented:
Eclair #1738 was a bit challenging to write, would love if someone could sanity-check me :-)

t-bast commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,182 @@
+  merchants and other users who want to begin receiving payments as
+  soon as the channel finishes opening.
+
+- [Eclair #1738][] corrects the penalizing of revoked HTLC transactions by
+  spending all outputs spendable by the revocation key instead of just the
+  first. This is to account for the possibility that [anchor outputs][topic
+  anchor outputs] enabled peers merge their
+  `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` HTLC transactions, resulting in an
+  aggregate transaction with multiple outputs to be penalized.
```
What about:

```suggestion
- [Eclair #1738][] updates the penalty mechanism for [anchor outputs][topic anchor outputs]
  revoked HTLC transactions. Anchor outputs introduces the possibility
  to merge multiple `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` HTLC transactions,
  resulting in an aggregate transaction with multiple outputs to be penalized at once.
  The PR ensures that all outputs that are spendable with the revocation key are claimed
  instead of claiming only one per transaction.
```

Thanks again for highlighting this PR, it could be useful for other implementers who may not have noticed this edge case.

jnewbery replied:
Change looks good to me. Thanks @t-bast!

harding replied:
Made a different edit that incorporates several of @t-bast's suggested changes (thanks!).  In past writing ([e.g.](https://bitcoinops.org/en/newsletters/2020/12/02/#proposed-intermediate-solution-for-ln-sighash-single-fee-theft)) I've tried not to refer to the sighash changes as anchor outputs, since it's independent of the changes made to use CPFP carve out---it just happened to have been made in the same protocol change as anchor outputs.  My edited version tries to make that distinction (and also doesn't backlink to this item from our anchor outputs topic for that reason) and also adds a link to a recent news item where we mentioned HTLC enforcement batching.

Thanks @dongcarl for the writeup and @t-bast for not only reviewing but also suggesting edits!

jnewbery commented:
SE section looks good. Thanks @bitschmidty!

harding commented:
Clean rebase on master to allow linking to the new dual funding topic.  Hopefully addressed all feedback.

Thanks @dongcarl, @jnewbery, and @bitschmidty for the items this week and @jonatack, @jnewbery, and @t-bast for reviews!  I thought the StackExchange section this month was especially good (and not at all because it linked to one of my answers).

bitschmidty reviewed (commented):
ACK 1f8860aae47a41f795bdcac77d52e2472fcfe12d

harding commented:
Added action item and release item for BTCPay Server security upgrade announcement; thanks @jonatack for notifying me about it!

adamjonas commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,186 @@
+- [Should block height or MTP or a mixture of both be used in a soft fork activation mechanism?]({{bse}}103854)
+  David A. Harding outlines the advantages and disadvantages of both Median
+  Time Past (MTP) and block heights as timing mechanisms within Bitcoin. MTP
+  roughly corresponds to clock time but can manipulated by miners to skip a
```
```suggestion
  roughly corresponds to clock time but can be manipulated by miners to skip a
```

jonatack commented:
Nice 👍 it's midnight here but I'll try to do another review pass tomorrow before noon UTC

harding commented:
Pushed edits for @jnewbery suggestion to mention batching post and @adamjonas typo find.  Thanks!

jonatack reviewed (commented):
Did another read-through and checked links.

jnewbery commented on _posts/en/newsletters/2021-03-31-newsletter.md:
```diff
@@ -0,0 +1,205 @@
+
+- **Upgrade BTCPay Server to 1.0.7.1:** this [release][btcpay server
+  1.0.7.1] fixes "one critical and several low-impact vulnerabilities
+  that affected BTCPay Server versions 1.0.7.0 and older", according to
```
See previous discussion on "programmer quotes:" https://github.com/bitcoinops/bitcoinops.github.io/pull/506#discussion_r560536416

bitschmidty commented:
Looking good team. Thanks @harding @jnewbery @dongcarl @jonatack @adamjonas @t-bast for your help this week!

jnewbery commented:
Mega edition this week. Thanks @harding et al!
